### ISSUE TITLE
reg_loss update error.

### ISSUE THREAD

**Issue by Zhuysheng** [JunxuanZhang/TURN-TAP-pytorch#1]
https://github.com/JunxuanZhang/TURN-TAP-pytorch/blob/50fd0d3944c58bcd52bbfe49f97d92c00c9fd16f/main.py#L139
when runs to this line, it shows Process finished with exit code 136(interrupted by signal 8: SIGFPE)， what's the problem?

**Comment by JunxuanZhang** [comment]
> https://github.com/JunxuanZhang/TURN-TAP-pytorch/blob/50fd0d3944c58bcd52bbfe49f97d92c00c9fd16f/main.py#L139
> 
> 
> when runs to this line, it shows Process finished with exit code 136(interrupted by signal 8: SIGFPE)， what's the problem?

I don't think this problem is caused by my code, maybe you can check your system enviroment.

**Comment by Zhuysheng** [comment]
@JunxuanZhang 
谢谢！我已经把环境设置到：
pytorch 0.4.0
cuda 8.0
python 2.7
运行到这一步时，还是出现问题：当第一个周期labels为0时，计算torch.sum(labels)，结果也为0，update参数时self.avg=self.sum/self.count，count也为0，会出现ZeroDivisionError。请问怎么解决。

**Comment by JunxuanZhang** [comment]
> @JunxuanZhang
> 谢谢！我已经把环境设置到：
> pytorch 0.4.0
> cuda 8.0
> python 2.7
> 运行到这一步时，还是出现问题：当第一个周期labels为0时，计算torch.sum(labels)，结果也为0，update参数时self.avg=self.sum/self.count，count也为0，会出现ZeroDivisionError。请问怎么解决。

这个问题可能是你的数据不够随机，一个batch里面全都是负样本，你可以对数据读取做一些操作，比如每个batch里面选取固定数目的正样本和负样本，保证正负都有，就不会出现问题，具体你尝试修改turn_dataset.py文件吧

**Comment by Zhuysheng** [comment]
@JunxuanZhang 谢谢！

**Comment by Zhuysheng** [comment]
还有个问题，请问 reg_start=round_reg_start/30，这边30是什么意思？

**Comment by JunxuanZhang** [comment]
> 还有个问题，请问 reg_start=round_reg_start/30，这边30是什么意思？

fps
